lib/widgets/breez_avatar_dialog.dart

```diff
@@ -11,147 +12,180 @@ import 'package:image/image.dart' as DartImage;
-                  stepGranularity: 0.1,
-                  group: _autoSizeGroup,
+                Expanded(
+                  child: FlatButton(
+                    padding: EdgeInsets.only(
+                      bottom: 20.0,
+                      top: 26.0,
+                    ),
+                    child: AutoSizeText(
+                      'GALLERY',
+                      style: theme.whiteButtonStyle,
+                      maxLines: 1,
+                      minFontSize: MinFontSize(context).minFontSize,
+                      stepGranularity: 0.1,
+                      group: _autoSizeGroup,
+                    ),
+                    onPressed: () {
+                      _pickImage(context, setState);
+                    },
+                  ),
                 ),
-                onPressed: () {
-                  _pickImage(context);
-                },
-              ),
+              ],
             ),
-          ],
+          ),
+        ]),
+        content: SingleChildScrollView(
+          child: ListBody(
+            children: <Widget>[
+              Theme(
+                data: ThemeData(
+                    primaryColor:
+                        Theme.of(context).primaryTextTheme.body1.color,
+                    hintColor: Theme.of(context).primaryTextTheme.body1.color),
+                child: TextField(
+                    style: Theme.of(context).primaryTextTheme.body1,
+                    controller: _nameInputController,
+                    decoration: InputDecoration(hintText: 'Enter your name'),
+                    onSubmitted: (text) {}),
+              )
+            ],
+          ),
         ),
-      ),
-    ]),
-    content: SingleChildScrollView(
-      child: ListBody(
-        children: <Widget>[
-          Theme(
-            data: ThemeData(
-                primaryColor: Theme.of(context).primaryTextTheme.body1.color,
-                hintColor: Theme.of(context).primaryTextTheme.body1.color),
-            child: TextField(
-                style: Theme.of(context).primaryTextTheme.body1,
-                controller: _nameInputController,
-                decoration: InputDecoration(hintText: 'Enter your name'),
-                onSubmitted: (text) {}),
-          )
+        actions: <Widget>[
+          FlatButton(
+            child: Text('CANCEL',
+                style: Theme.of(context).primaryTextTheme.button),
+            onPressed: () {
+              Navigator.of(context).pop();
+            },
+          ),
+          FlatButton(
+            child:
+                Text('SAVE', style: Theme.of(context).primaryTextTheme.button),
+            onPressed: () async {
+              if (_imageData != null) {
+                var action = UploadProfilePicture(_imageData);
+                userBloc.userActionsSink.add(action);
+                await action.future.then((_) async {
+                  Navigator.of(context).pop();
```

Comment:
This block is a bit tricky since lots of things can happen during the upload (like user cancel the dialog), the upload will continue in the background and when completed the navigator will pop the current route without checking it is the right route.
Also I suggest to add some UI indication (loading) during the upload.

Reply:
[Prevented user to close dialog or cancel during upload](https://github.com/breez/breezmobile/pull/283/commits/fcb442c9e84dc2aeb449fc677c833418f9eb73b3)